docs/rfcs/0001-rfc-process.md

```diff
@@ -0,0 +1,63 @@
+- Status: Proposed
+- Date: 2023-05-23
+- RFC PR: TODO
+
+
+## Summary
+Introducing a process for proposing, reviewing, and documenting decisions with high-impact implications for our codebase.
+
+## Motivation
+
+RFCs authored by this team are currently submitted and reviewed in a number of places, namely Confluence and Google Docs.
+
+Several challenges have been noted with this practice:
+- Docs are accessible to Mozillians only
+- Platform constraints make asynchronous discussions difficult and impossible to reference later (Confluence) 
+- Uneven awareness and ability for interested parties to participate in decisions as they're being made (Google Docs) 
+
+We need a more inclusive, more collaborative process for making decisions that impact our codebase.
+
+## Proposed Solution
+
+RFCs will be submitted and reviewed as pull requests in this repository. If approved, they'll be stored as documentation.
+
+**Salient points in favor**
+ - Viewable by all, not just Mozilla-internals
+ - Closer to the code, easier to find
+ - In keeping with how other teams do it (TODO link evidence)
+
+
+## Guide-level explanation
+
+The high-level process to create an RFC is as follows:
+
+* Create an RFC document (like this one). (TODO: templates, maybe)
+* Open a pull request for the RFC document.
+* Ask for feedback on the pull request.
+
+During the lifetime of an RFC:
+
+* Discussions happen asynchronously on the pull request; anyone may engage.
+* Build consensus (TODO: agreement?) and integrate feedback.
+
+After the discussion phase has concluded:
+
+* If consensus(TODO: agreement?) is reached, the RFC is considered "accepted" and merged into the repository for documentation purposes.
+* If consensus(TODO: agreement?) is not reached, the RFC is considered "rejected" and the pull request is closed. Rejected RFC proposals may be revived should the requirements change in the future.
+
+Authors may implement and submit the feature as a pull request once the RFC is accepted.
+
+## Drawbacks
+- (?)
+
+## Questions
+- About nomenclature: Are we ok calling these 'RFCs'? Firefox Monitor and Firefox Relay refer to these as ADRs but 'RFC' strikes me as more to the point.
+- About templates: I'm in favor of including an RFC template as an example but I'd prefer we not be overly prescriptive about it, at least early on.
+- About documentation living elsewhere: What to do about design documents currently housed in Confluence, if anything? Are there documents we feel must be migrated over to GitHub and stored as RFCs? (Does not require a decision now) 
```

Comment:
I think that would be unnecessary.

Reply:
We have 23 tech specs listed in Confluence, I like the idea of having everything in one place for future references. I can move them over if we decide to do that, it should be fairly quick

Reply:
I think we should consider these on an individual basis. If they are essential/helpful, I'd recommend we bring them over as documentation but perhaps not as RFCs. I do think it would be helpful to add links to future RFC documents to the list in Confluence, as we do for a number of the documents still living in Google Docs.